src/protocol.rs

```diff
@@ -407,6 +416,132 @@ mod tests {
     use std::collections::HashMap;
     use tracing::debug;
 
+    //#[cfg_attr(feature = "flame_it", flame)]
+    #[test]
+    fn process_single_message_wrong_session_id() -> Result<()> {
+        let mut rng = init_testing();
+        let QUORUM_SIZE = 3;
+        // Set GLOBAL config for participants
+        let configs = ParticipantConfig::random_quorum(QUORUM_SIZE, &mut rng);
+
+        // Set up auxinfo participants
+        let auxinfo_sid = Identifier::random(&mut rng);
+        let mut auxinfo_quorum = configs
+            .iter()
+            .map(|config| Participant::<AuxInfoParticipant>::from_config(config, auxinfo_sid, ()))
+            .collect::<Vec<_>>();
+
+        let message_type = MessageType::Auxinfo(crate::messages::AuxinfoMessageType::Public);
+        let mut rng1 = init_testing();
+        let identifier = Identifier::random(&mut rng1);
+        let from = auxinfo_quorum[0].id;
+        let to = auxinfo_quorum[0].id;
+        let unverified_bytes = vec![0x01, 0x02, 0x03];
+        let message = Message::new(message_type, identifier, from, to, &unverified_bytes);
+        let participant = auxinfo_quorum.iter_mut().choose(&mut rng).unwrap();
+
+        if let Err(e) = participant.process_single_message(&message, &mut rng) {
+            assert_eq!(
+                e,
+                InternalError::CallingApplicationMistake(CallerError::WrongSessionId)
+            );
+        };
```

Comment:
todo: this test will actually pass if `process_single_message` doesn't throw an error, since it'll skip this condition entirely and hit `Ok(())`. Consider breaking this into two asserts (is it an error? is it the right error?) like:
```rust
let result = participant.process_single_message(...);
assert!(result.is_err());
assert_eq!(result.unwrap_err(), InternalError::CallingApplicationMistake(CallerError::WrongSessionId));